### ISSUE TITLE
Mac M1 - instalation error with pnpm - ERR 404 even doing brew instalation

### ISSUE THREAD

**Issue by Donivanes** [Automattic/node-canvas#2431]
## Issue or Feature
- [x] If this is an issue with installation, I have read the [troubleshooting guide](https://github.com/Automattic/node-canvas/issues/1511).

After doing `pnpm i canvas` i have this issue 

```
Progress: resolved 2012, reused 1945, downloaded 0, added 0, done
node_modules/.pnpm/canvas@2.11.2/node_modules/canvas: Running install script, failed in 1.6s
.../canvas@2.11.2/node_modules/canvas install$ node-pre-gyp install --fallback-to-build --update-binary
│ node-pre-gyp info it worked if it ends with ok
│ node-pre-gyp info using node-pre-gyp@1.0.11
│ node-pre-gyp info using node@20.7.0 | darwin | arm64
│ node-pre-gyp http GET https://github.com/Automattic/node-canvas/releases/download/v2.11.2/canvas-v2.11.2-node-v115-darwin-unknown-arm64.tar.gz
│ node-pre-gyp ERR! install response status 404 Not Found on https://github.com/Automattic/node-canvas/releases/download/v2.11.2/canvas-v2.11.2-node-v1…
│ node-pre-gyp WARN Pre-built binaries not installable for canvas@2.11.2 and node@20.7.0 (node-v115 ABI, unknown) (falling back to source compile with …
│ node-pre-gyp WARN Hit error response status 404 Not Found on https://github.com/Automattic/node-canvas/releases/download/v2.11.2/canvas-v2.11.2-node-…
│ gyp info it worked if it ends with ok
│ gyp info using node-gyp@10.1.0
│ gyp info using node@20.7.0 | darwin | arm64
│ gyp info ok
│ gyp info it worked if it ends with ok
│ gyp info using node-gyp@10.1.0
│ gyp info using node@20.7.0 | darwin | arm64
│ gyp info find Python using Python version 3.12.6 found at "/opt/homebrew/opt/python@3.12/bin/python3.12"
│ gyp info spawn /opt/homebrew/opt/python@3.12/bin/python3.12
│ gyp info spawn args [
│ gyp info spawn args '/Users/ivan/Library/pnpm/.tools/pnpm/9.10.0/node_modules/.pnpm/pnpm@9.10.0/node_modules/pnpm/dist/node_modules/node-gyp/gyp/gyp_…
│ gyp info spawn args 'binding.gyp',
│ gyp info spawn args '-f',
│ gyp info spawn args 'make',
│ gyp info spawn args '-I',
│ gyp info spawn args '-I',
│ gyp info spawn args '/Users/ivan/Library/pnpm/.tools/pnpm/9.10.0/node_modules/.pnpm/pnpm@9.10.0/node_modules/pnpm/dist/node_modules/node-gyp/addon.gy…
│ gyp info spawn args '-I',
│ gyp info spawn args '/Users/ivan/Library/Caches/node-gyp/20.7.0/include/node/common.gypi',
│ gyp info spawn args '-Dlibrary=shared_library',
│ gyp info spawn args '-Dvisibility=default',
│ gyp info spawn args '-Dnode_engine=v8',
│ gyp info spawn args '--depth=.',
│ gyp info spawn args '--no-parallel',
│ gyp info spawn args '--generator-output',
│ gyp info spawn args 'build',
│ gyp info spawn args '-Goutput_dir=.'
│ gyp info spawn args ]
│ gyp info ok
│ gyp info it worked if it ends with ok
│ gyp info using node-gyp@10.1.0
│ gyp info using node@20.7.0 | darwin | arm64
│ gyp info spawn make
│ gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
│   SOLINK_MODULE(target) Release/canvas-postbuild.node
│   CXX(target) Release/obj.target/canvas/src/backend/Backend.o
│ In file included from ../src/backend/Backend.cc:1:
│ ../src/backend/Backend.h:5:10: fatal error: 'exception' file not found
│     5 | #include <exception>
│       |          ^~~~~~~~~~~
│ 1 error generated.
│ make: *** [Release/obj.target/canvas/src/backend/Backend.o] Error 1
│ gyp ERR! build error
│ gyp ERR! stack Error: `make` failed with exit code: 2
│ gyp ERR! stack at ChildProcess.<anonymous> (/Users/ivan/Library/pnpm/.tools/pnpm/9.10.0/node_modules/.pnpm/pnpm@9.10.0/node_modules/pnpm/dist/node_mo…
│ gyp ERR! System Darwin 24.0.0
│ gyp ERR! command "/Users/ivan/.volta/tools/image/node/20.7.0/bin/node" "/Users/ivan/Library/pnpm/.tools/pnpm/9.10.0/node_modules/.pnpm/pnpm@9.10.0/no…
│ gyp ERR! node -v v20.7.0
│ gyp ERR! node-gyp -v v10.1.0
│ gyp ERR! not ok
│ node-pre-gyp ERR! build error
│ node-pre-gyp ERR! stack Error: Failed to execute '/Users/ivan/.volta/tools/image/node/20.7.0/bin/node /Users/ivan/Library/pnpm/.tools/pnpm/9.10.0/nod…
│ node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/Users/xxxxxx/node_modules/.pnpm/@mapbox+node-pre-gyp…
│ node-pre-gyp ERR! stack     at ChildProcess.emit (node:events:514:28)
│ node-pre-gyp ERR! stack     at maybeClose (node:internal/child_process:1105:16)
│ node-pre-gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:305:5)
│ node-pre-gyp ERR! System Darwin 24.0.0
│ node-pre-gyp ERR! command "/Users/ivan/.volta/tools/image/node/20.7.0/bin/node" "/Users/xxxxx/node_modules/.p…
│ node-pre-gyp ERR! cwd /Users/ixxxxx/node_modules/.pnpm/canvas@2.11.2/node_modules/canvas
│ node-pre-gyp ERR! node -v v20.7.0
│ node-pre-gyp ERR! node-pre-gyp -v v1.0.11
│ node-pre-gyp ERR! not ok
│ Failed to execute '/Users/ivan/.volta/tools/image/node/20.7.0/bin/node /Users/ivan/Library/pnpm/.tools/pnpm/9.10.0/node_modules/.pnpm/pnpm@9.10.0/nod…
└─ Failed in 1.6s at /Users/xxxx/.pnpm/canvas@2.11.2/node_modules/canvas
 ELIFECYCLE  Command failed with exit code 1.
```

## Steps to Reproduce

Just doing `pnpm i canvas`

I have also done the brew instalation because I'm using Mac M1 

`brew install pkg-config cairo pango libpng jpeg giflib librsvg pixman python-setuptools`

This doesn't fix it

## Your Environment
* Version of node-canvas `2.11.2`
* Environment `node: 20.7.0  Mac M1 verion 15`

**Comment by jialu98** [comment]
Same with node v20.17 and Mac M2 macOS 15.0

**Comment by harrisonhowardax** [comment]
Happening on `v18.19.0` Mac M1 macos 15.0

**Comment by harrisonhowardax** [comment]
Okay guys I found out what was breaking. For me it was Xcode command line tools.
Do `xcode-select --print-path`, I deleted whatever the path was.
When you run `xcode-select --print-path` it should show a completely different path.
Run `gcc` after deleting the broken one and confirm it's still working.

Once I did that I restarted my computer and ran the install again, worked like a charm. Let me know if this fixes anyone else's issue.

**Comment by nicorch** [comment]
> Okay guys I found out what was breaking mine. For me, it was Xcode command line tools. Do `xcode-select --print-path`, I deleted whatever the path was. When you run `xcode-select --print-path` it should show a different path. Run `gcc` after deleting the broken one and confirm it's still working.
> 
> Once I did that I restarted my computer and ran the install again, worked like a charm. Let me know if this fixes anyone else's issue.
> 
> EDIT: If the path that is reported is `/Library/Developer/CommandLineTools` then it's wrong. The expected path is `/Applications/Xcode.app/Contents/Developer`. The MacOS 15 update is likely the reason this occurs.

Worked for me, Thanks

**Comment by Donivanes** [comment]
> /Xcode.app



> Okay guys I found out what was breaking mine. For me, it was Xcode command line tools. Do `xcode-select --print-path`, I deleted whatever the path was. When you run `xcode-select --print-path` it should show a different path. Run `gcc` after deleting the broken one and confirm it's still working.
> 
> Once I did that I restarted my computer and ran the install again, worked like a charm. Let me know if this fixes anyone else's issue.
> 
> EDIT: If the path that is reported is `/Library/Developer/CommandLineTools` then it's wrong. The expected path is `/Applications/Xcode.app/Contents/Developer`. The MacOS 15 update is likely the reason this occurs.

It worked for me also. Thank you. 

Even when I do `xcode-select --install` it was telling me that I have already installed the command line tools, I had to go to the appStore to install Xcode manually (because the path `/Applications/Xcode.app/Contents/Developer` did not exist). After doing that, the path you've mentioned changed automatically and the installation worked.

**Comment by Donivanes** [comment]
Closing the issue. @harrisonhowardax thank so much you again 👏🏽

**Comment by jedrzejjurek** [comment]
after xcode manual installation and agreeing to license, I had to reboot again, then it worked flawlessly 👌 after hours of research that finally helped, thanks @harrisonhowardax

**Comment by FominSergiy** [comment]
Chime-in here as well - I just deleted and reinstalled xcode tools and that worked for me.

```bash
sudo rm -rf /Library/Developer/CommandLineTools
xcode-select --install 
```

**Comment by pablo-lopes-smava** [comment]
Thanks @harrisonhowardax . 
I have the Mac M2 with OS v15.
First of all, I've updated the XCode to the new version. The path `/Library/Developer/CommandLineTools` didn't not change.
So, I uninstalled the xcode command line tools and reinstall following the comments from @FominSergiy . But the path didn't change.
Even though, with the new version of XCode and xcode command line tools, now it's working.
